modules/core/react/lib/InputProvider.tsx

```diff
@@ -4,6 +4,7 @@ import elementClosestPolyfill from 'element-closest';
 
 export interface InputProviderProps {
   provideIntent?: boolean;
+  container?: HTMLElement;
```

Comment:
This will have to accept refs as well, otherwise we ask react apps to use `findDOMNode` (which is being deprecated)

Reply:
A ref doesn't require `findDOMNode`, it will have access to the DOM node in its `.current` property.

https://reactjs.org/docs/hooks-reference.html#useref

We could add support for a `ref` as a convenience for React apps:

```tsx
const App = () => {
  const ref = React.useRef()
  return (
    <div ref={ref}>
      <InputProvider container={ref} />
      {/* other children */}
    </div>
  )
}
```

We could update the signature to accept `HTMLElement | React.RefObject<HTMLElement>` and do detection inside `InputProvider`

Reply:
You are right, sorry. I guess I was wanting to avoid them having to do `container={ref.current}`